extensions/smallrye-reactive-messaging/runtime/src/main/java/io/quarkus/smallrye/reactivemessaging/runtime/devmode/DevModeSupportConnectorFactoryInterceptor.java

```diff
@@ -0,0 +1,82 @@
+package io.quarkus.smallrye.reactivemessaging.runtime.devmode;
+
+import java.util.concurrent.CompletableFuture;
+import java.util.function.Supplier;
+
+import javax.annotation.Priority;
+import javax.interceptor.AroundInvoke;
+import javax.interceptor.Interceptor;
+import javax.interceptor.InvocationContext;
+
+import org.eclipse.microprofile.reactive.messaging.Message;
+import org.eclipse.microprofile.reactive.streams.operators.PublisherBuilder;
+import org.eclipse.microprofile.reactive.streams.operators.ReactiveStreams;
+import org.eclipse.microprofile.reactive.streams.operators.SubscriberBuilder;
+import org.reactivestreams.Subscriber;
+import org.reactivestreams.Subscription;
+
+@Interceptor
+@DevModeSupportConnectorFactory
+@Priority(Interceptor.Priority.PLATFORM_BEFORE + 10)
+public class DevModeSupportConnectorFactoryInterceptor {
+    private static Supplier<CompletableFuture<Boolean>> onMessage;
```

Comment:
This should have been volatile

Reply:
Ah that is true. I copied this pattern from other dev mode handlers, without too much thinking. I'll submit a PR.

Reply:
Done: #16263